posix/unix.c

```diff
@@ -237,15 +252,34 @@ static unixsock_t *unixsock_get_connected(unixsock_t *s)
 
 static void unixsock_put(unixsock_t *s)
 {
+	unixsock_t *r;
+
 	proc_lockSet(&unix_common.lock);
 	if (--s->refs <= 0) {
```

Comment:
As you're already on fixing MISRA: statements should not have side effects

```suggestion
    s->refs--;
	if (s->refs <= 0) {
```